factory/usr/lib/systemd/system-generators/kernel-snap-generator

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+set -eu
+
+: "${KERNEL_MNT_POINT:=/run/mnt/kernel}"
+
+if ! mountpoint -q "${KERNEL_MNT_POINT}"; then
+    exit 0
+fi
+
+# FIXME use SYSTEMD_IN_INITRD when using systemd 251+
+if [ -f /etc/initrd-release ]; then
+    sysroot=/sysroot
+    sysroot_unit=sysroot
+    target=initrd-fs.target
+else
+    sysroot=
+    sysroot_unit=
+    target=local-fs.target
+fi
+
+: "${OS_RELEASE:="${sysroot}/etc/os-release"}"
+
+if ! grep -q "^ID=ubuntu$" "${OS_RELEASE}"; then
```

Comment:
The file won't exist when this is run from initramfs, is that expected?

Reply:
I have commented.